### PR TITLE
ARROW-5534: [GLib] Add garrow_table_concatenate()

### PR DESCRIPTION
**Pull request by shiro615**

### CODE REVIEW

File: c_glib/arrow-glib/table.h [hunk 1]
```diff
@@ -89,5 +89,8 @@ GArrowTable    *garrow_table_replace_column(GArrowTable *table,
                                             GError **error);
 gchar          *garrow_table_to_string     (GArrowTable *table,
                                             GError **error);
+GArrowTable *garrow_table_concatenate(GArrowTable *table,
```
Comment (kou) [inline c_glib/arrow-glib/table.h]: Could you add `GARROW_AVAILABLE_IN_0_14`?

**Review by kou — CHANGES_REQUESTED** [review]
Thanks!
We can merge this with a small fix.

File: c_glib/arrow-glib/table.h (hunk 1 above)
Comment (shiro615) [reply c_glib/arrow-glib/table.h]: Yes. I've added it.

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**